_resources/port1.0/group/crossgdb-1.0.tcl

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+#
+# This PortGroup automatically sets all the fields of the various cross gdb
+# ports (e.g. arm-none-eabi-gdb).
+#
+# Usage:
+#
+#   PortGroup           crossgdb 1.0
+#   crossgdb.setup      arm-none-eabi 11.1
+
+options crossgdb.target
+
+array set crossgdb.version_info {
+    11.1 {xz {
+        rmd160  da07d7504be6c7d11c98751e0cbef75685daceba \
+        sha256  cccfcc407b20d343fb320d4a9a2110776dd3165118ffd41f4b1b162340333f94 \
+        size    22040696
+    }}
+}
+
+proc crossgdb.setup {target version} {
+    global crossgdb.target crossgdb.version crossgdb.version_info
+
+    set crossgdb.target ${target}
+    set crossgdb.version ${version}
+
+    uplevel {
+        PortGroup           compiler_blacklist_versions 1.0
+        name                ${crossgdb.target}-gdb
+        version             ${crossgdb.version}
+        categories          cross devel
+        platforms           darwin
+        license             GPL-3+
+        maintainers         nomaintainer
+
+        description         GDB: The GNU Project Debugger for ${crossgdb.target}
+        long_description    {*}${description}
+
+        homepage            https://www.gnu.org/software/gdb/
+        master_sites        gnu:gdb
+
+        if {[info exists crossgdb.version_info(${version})]} {
+            use_[lindex [set crossgdb.version_info(${version})] 0] yes
+
+            checksums gdb-${version}${extract.suffix} {*}[lindex [set crossgdb.version_info(${version})] 1]
+        } else {
+            # the old default
+            use_bzip2       yes
+        }
+
+        dist_subdir         gdb[lindex [split ${version} .] 0]
+        distfiles           gdb-${version}${extract.suffix}:gdb
+        
+        worksrcdir          gdb-${version}
+
+        # These dependencies are listed under depends_lib rather than
+        # depends_build because gdb will link with libraries they provide if
+        # installed. There may be more. See variable host_libs in configure.ac.
+        # port:guile should also be a dependency, but currently does not build
+        # universally.
+        # https://trac.macports.org/ticket/48767
+        depends_lib         port:${crossgdb.target}-gcc \
```

Comment:
not sure when this `proc` will get executed, but if it happens after possible dependencies in a Portfile are defined this will wipe those out. Perhaps one should use `depends_lib-append` here to be safe?

Reply:
can they also be sorted alphabetically?

Reply:
Sure, done